Server/mods/deathmatch/logic/luadefs/CLuaACLDefs.cpp

```diff
@@ -1049,3 +1051,37 @@ int CLuaACLDefs::OOP_isObjectInACLGroup(lua_State* luaVM)
     lua_pushboolean(luaVM, false);
     return 1;
 }
+
+std::variant<std::vector<CAccessControlListGroup*>, bool>
+CLuaACLDefs::aclObjectGetGroups(std::string strObject)
+{
+    CAccessControlListGroupObject::EObjectType objectType;
+    const char* szObjectAfterDot = strObject.c_str();
+    if (StringBeginsWith(szObjectAfterDot, "resource."))
+    {
+        szObjectAfterDot += 9;
+        objectType = CAccessControlListGroupObject::OBJECT_TYPE_RESOURCE;
+    }
+    else if (StringBeginsWith(szObjectAfterDot, "user."))
+    {
+        szObjectAfterDot += 5;
+        objectType = CAccessControlListGroupObject::OBJECT_TYPE_USER;
+    }
+    else
+    {
+        return false;
```

Comment:
This function should never anything else other than a list of groups. Throw an exception here.